avocado/utils/cpu.py

```diff
@@ -188,3 +189,56 @@ def offline(cpu):
     if _get_cpu_status(cpu):
         return 1
     return 0
+
+
+def get_cpu_idle():
+    """
+    Get current cpu idle values
+    """
+    cpus_list = cpu_online_list()
+    states = range(len(glob.glob("/sys/devices/system/cpu/cpu0/cpuidle/state*")))
+    cpu_idle_state = [[0]*len(states)]*len(cpus_list)
+    for c in cpus_list:
+        for state in states:
+            state_file = "/sys/devices/system/cpu/cpu%s/cpuidle/state%s/disable" % (c, state)
+            try:
+                cpu_idle_state[c][state] = int(open(state_file).read())
+            except Exception:
```

Comment:
Well, on symetrical systems it should not fail and on heterogenous systems you might not get all states. Let's use system with 2 cpus with 3 states and 2 cpus with 6 states. This code returns:

```python
[[0, 0, 0], [0, 0, 0], [0, 0, 0], [0, 0, 0]]
```

 If you want to support heterogenous systems I'd suggest changing the logic to always get `state*`.

Think to reconsideration is whether the matrix should be symetrical:

```python
[[0, 0, 0 0, 0, 0], [0, 0, 0 0, 0, 0], [0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0]]
```

I'd rather go with sparse one so with 2 cpus with 3 states and 2 cpus with 6 states it'd return somthing like:

```python
[[0, 0, 0], [0, 0, 0], [0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0]]
```

Reply:
PS: Either way don't catch generic `Exception`... `IOError` should suffice for non-existing files...

Reply:
If that shouldn't be supported I'd suggest not masking the failure. Then we'll get a report when some user use it on such system.

Reply:
Sure, That make sense, will do so